compiler/rustc_trait_selection/src/solve/fulfill/derive_errors.rs

```diff
@@ -390,12 +453,6 @@ impl<'tcx> ProofTreeVisitor<'tcx> for BestObligation<'tcx> {
                 }
             }
 
-            // Skip nested goals that aren't the *reason* for our goal's failure.
-            match (self.consider_ambiguities, nested_goal.result()) {
-                (true, Ok(Certainty::Maybe(MaybeCause::Ambiguity))) | (false, Err(_)) => {}
-                _ => continue,
-            }
```

Comment:
we now skip them when recursing as we also need this behavior when recursing anywhere else in this function